### PR TITLE
fix(browser-tx): fix estimate gas and send TX on the browser

### LINKED ISSUE #4557: [base_bc] gas estimation is not working
### Description

1. enable browser
2. switch to ropsten
3. open simpledapp.eth
4. send a transaction

```
qrc:/imports/shared/controls/GasSelector.qml:76: Error: Cannot assign [undefined] to double
qrc:/imports/shared/popups/SignTransactionModal.qml:95: Error: Cannot assign to non-existent property "title"
qrc:/StatusQ/src/StatusQ/Popups/statusModal/StatusModalFooter.qml:57:9: QML Row: Cannot specify left, right, horizontalCenter, fill or centerIn anchors for items inside Row. Row will not function.
qrc:/imports/shared/controls/GasSelector.qml:76: Error: Cannot assign [undefined] to double
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double
qrc:/StatusQ/src/StatusQ/Popups/statusModal/StatusModalFooter.qml:57:9: QML Row: Cannot specify left, right, horizontalCenter, fill or centerIn anchors for items inside Row. Row will not function.
qrc:/imports/shared/popups/SignTransactionModal.qml:95: Error: Cannot assign to non-existent property "title"
qrc:/StatusQ/src/StatusQ/Popups/statusModal/StatusModalFooter.qml:57:9: QML Row: Cannot specify left, right, horizontalCenter, fill or centerIn anchors for items inside Row. Row will not function.
```

Screenshot 2022-01-25 at 10 38 56

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4557

This makes the browser send transaction work. Hooks the estimateGas and also implements the sendTransaction for the provider

### CODE REVIEW

**Comment by anastasiyaig** [comment]
@jrainville 
it is not working to me:

```
core.nim:24 rpc_method=wallet_getTransfersByAddress
qrc:/imports/shared/controls/GasSelector.qml:76: Error: Cannot assign [undefined] to double
qrc:/imports/shared/controls/GasSelector.qml:76: Error: Cannot assign [undefined] to double
qrc:/imports/shared/popups/SendModal.qml:138:17: Unable to assign [undefined] to QString
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double
qrc:/imports/shared/controls/chat/GasSelectorButton.qml:34:5: Unable to assign [undefined] to double

```

**Comment by richard-ramos** [comment]
In this screenshot that @anastasiyaig shared, looks like EIP1559 behavior is not being applied. It should display two fields for fees instead of only the gas price

**Comment by anastasiyaig** [comment]
can we fix it please please? :)

**Comment by anastasiyaig** [comment]
alsi @jrainville i cant confirm transaction in dapp, the button is trying to run away

Screenshot 2022-01-28 at 16 21 17

**Comment by jrainville** [comment]
@anastasiyaig I fixed the button position and the Send modal gas problem.
The Send itself won't work, because it requires more work, so it'd be better to tackle it in another PR

**Comment by anastasiyaig** [comment]
@jrainville 

is there anything we can do to improve it still? The app has become so slow and it even crashed (when clicking through wallet)

```
goroutine 4483 [select]:
net/http.(*http2ClientConn).roundTrip(0x1c0001e0f00, 0x1c0020ad600)
	/usr/local/go/src/net/http/h2_bundle.go:7771 +0xcba
net/http.(*http2Transport).RoundTripOpt(0x1c000030770, 0x1c0020ad600, {0x60})
	/usr/local/go/src/net/http/h2_bundle.go:7094 +0x1b7
net/http.(*http2Transport).RoundTrip(...)
	/usr/local/go/src/net/http/h2_bundle.go:7055
net/http.http2noDialH2RoundTripper.RoundTrip({0x113fd4780}, 0x1c0020ad600)
	/usr/local/go/src/net/http/h2_bundle.go:9312 +0x1b
net/http.(*Transport).roundTrip(0x113fd4780, 0x1c0020ad500)
	/usr/local/go/src/net/http/transport.go:539 +0x390
net/http.(*Transport).RoundTrip(0x30, 0x1133c7aa0)
	/usr/local/go/src/net/http/roundtrip.go:18 +0x19
net/http.send(0x1c0020ad500, {0x1133c7aa0, 0x113fd4780}, {0x113367b40, 0x1, 0x0})
	/usr/local/go/src/net/http/client.go:252 +0x5d8
net/http.(*Client).send(0x1c00039d4a0, 0x1c0020ad500, {0x10, 0x207000, 0x0})
	/usr/local/go/src/net/http/client.go:176 +0x9b
net/http.(*Client).do(0x1c00039d4a0, 0x1c0020ad500)
	/usr/local/go/src/net/http/client.go:725 +0x908
net/http.(*Client).Do(...)
	/usr/local/go/src/net/http/client.go:593
github.com/ethereum/go-ethereum/rpc.(*httpConn).doRequest(0x1c0005c88c0, {0x1133ef2b8, 0x1c0021947c0}, {0x1132d82c0, 0x1c0019e7420})
	/Users/jenkins/go/pkg/mod/github.com/status-im/go-ethereum@v1.10.4-status.3/rpc/http.go:184 +0x21a
github.com/ethereum/go-ethereum/rpc.(*Client).sendHTTP(0x173344cd8, {0x1133ef2b8, 0x1c0021947c0}, 0x1c0020eae40, {0x1132d82c0, 0x1c0019e7420})
	/Users/jenkins/go/pkg/mod/github.com/status-im/go-ethereum@v1.10.4-status.3/rpc/http.go:136 +0x8b
github.com/ethereum/go-ethereum/rpc.(*Client).CallContext(0x1c000333200, {0x1133ef2b8, 0x1c0021947c0}, {0x1130bed40, 0x1c0001b1d20}, {0x112cbd5a6, 0x14}, {0x1c0005fc6c0, 0x2, 0x2})
	/Users/jenkins/go/pkg/mod/github.com/status-im/go-ethereum@v1.10.4-status.3/rpc/client.go:302 +0x208
github.com/ethereum/go-ethereum/ethclient.(*Client).HeaderByNumber(0x1c00059a6c0, {0x1133ef2b8, 0x1c0021947c0}, 0x6)
	/Users/jenkins/go/pkg/mod/github.com/status-im/go-ethereum@v1.10.4-status.3/ethclient/ethclient.go:182 +0x12b
github.com/status-im/status-go/services/wallet/chain.(*Client).HeaderByNumber(0x1c001ab8b20, {0x1133ef2b8, 0x1c0021947c0}, 0x4)
	/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/vendor/status-lib/vendor/status-go/services/wallet/chain/client.go:54 +0x55
github.com/status-im/status-go/services/wallet/transfer.checkRanges.func1({0x1133ef2b8, 0x1c0021947c0})
	/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/vendor/status-lib/vendor/status-go/services/wallet/transfer/concurrent.go:135 +0x479
github.com/status-im/status-go/services/wallet/async.(*AtomicGroup).Add.func1()
	/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/vendor/status-lib/vendor/status-go/services/wallet/async/async.go:114 +0x68
created by github.com/status-im/status-go/services/wallet/async.(*AtomicGroup).Add
	/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/vendor/status-lib/vendor/status-go/services/wallet/async/async.go:112 +0x92
QObject::~QObject: Timers cannot be stopped from another thread
Traceback (most recent call last)
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/src/nim_status_client.nim(146) nim_status_client
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/src/nim_status_client.nim(143) mainProc
/Users/jenkins/workspace/s-desktop_branches_macos_PR-4605/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

**Comment by jrainville** [comment]
That's a weird issue. It seems to have crashed in status-go. Do you have any idea which actions you took that caused that crash @anastasiyaig ?

Also, I can look some more, but I mostly focused on the browser in this PR. Is base_bc having the same problem with the wallet?
If yes, I'd say it's better to do it in another issue/PR.
If no, then I introduced those problems, and I will gladly fix them straight away

**Comment by anastasiyaig** [comment]
@jrainville basically i haven done anything specific, i pasted an address and noticed huge delays in the app responsiveness. I dont see it in base bc now
@iurimatias would u mind to take a look if u have time?

File: src/app_service/service/ens/utils.nim [hunk 1]
```diff
@@ -186,8 +186,16 @@ proc owner*(username: string): string =
...
+proc buildTransaction*(
+    source: Address,
+    value: Uint256,
+    gas = "",
+    gasPrice = "",
+    isEIP1559Enabled = false, 
+    maxPriorityFeePerGas = "",
+    maxFeePerGas = "",
+    data = ""
+    ): TransactionDataDto =
```
Comment (alaibe) [inline src/app_service/service/ens/utils.nim:198]: nit: fix indentation (2 spaces, not 4)

```
proc buildTransaction*(
  source: Address,
  value: Uint256,
  gas = "",
  gasPrice = "",
  isEIP1559Enabled = false, 
  maxPriorityFeePerGas = "",
  maxFeePerGas = "",
  data = ""
): TransactionDataDto =
```

File: ui/imports/shared/controls/CopyToClipBoardButton.qml [hunk 2]
```diff
@@ -59,7 +59,8 @@ Rectangle {
         onClicked: {
             if (textToCopy) {
                 // Not Refactored Yet
```
Comment (alaibe) [inline ui/imports/shared/controls/CopyToClipBoardButton.qml:61]: it is refactored now?

File: ui/imports/shared/controls/CopyToClipBoardButton.qml [hunk 3]
```diff
@@ -59,7 +59,8 @@ Rectangle {
...
-                //RootStore.copyToClipboard(textToCopy)
+                // TODO use a store instead
+                globalUtils.copyToClipboard(textToCopy)
```
Comment (alaibe) [inline ui/imports/shared/controls/CopyToClipBoardButton.qml:63]: I think there is copyToClipboard available in shared store

File: ui/imports/shared/popups/SendModal.qml [hunk 4]
```diff
@@ -146,35 +142,38 @@ ModalPopup {
...
-//                getFiatValue: RootStore.fiatValue
-//                defaultCurrency: RootStore.defaultCurrency
+                gasPrice: parseFloat(root.store.gasPrice)
+                getGasEthValue: root.store.getGasEthValue
+                getFiatValue: root.store.getFiatValue
+                defaultCurrency: root.store.currentCurrency
 
                 width: stack.width
                 property var estimateGas: Backpressure.debounce(gasSelector, 600, function() {
                       // Not Refactored Yet
```
Comment (alaibe) [inline ui/imports/shared/popups/SendModal.qml:152]: still valid?

**Review by alaibe — APPROVED** [review]
Nice, left a few comments

**Comment by jrainville** [comment]
Rebased. @richard-ramos can I get that sweet review.

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**